org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/repositories/actions/OpenRepositoryAction.java

```diff
@@ -36,20 +43,50 @@ public OpenRepositoryAction(IViewPart view) {
 
 	@Override
 	public void run() {
+
 		IRepository repository = getRepository();
 		if (repository != null) {
 			try {
-				// Open default external browser
-				PlatformUI.getWorkbench().getBrowserSupport().getExternalBrowser().openURL(new URL(repository.getUrl()));
+				// Get Repository link
+				String repoLink = getLink(repository);
+				// Open the link in default external browser
+				URI uri = new URI(repoLink);
+				PlatformUI.getWorkbench().getBrowserSupport().getExternalBrowser().openURL(uri.toURL());
 			} catch (PartInitException | MalformedURLException exception) {
 				MessageDialog.openError(this.view.getViewSite().getShell(), Messages.AbapGitView_action_open_repo_error_dialog_title,
 						exception.getMessage());
 				AbapGitUIPlugin.getDefault().getLog()
 						.log(new Status(IStatus.ERROR, AbapGitUIPlugin.PLUGIN_ID, exception.getMessage(), exception));
+			} catch (URISyntaxException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
```

Comment:
Yes, took reference from existing code

Reply:
You can avoid the duplicate code. In case the handling for an exception is different, only in that case add another catch block, otherwise handle the exception in the existing catch block.

So you could simple move the exception to the existing catch block.

Reply:
Make sense! will follow this way